src/api/procedures/utils.ts

```diff
@@ -322,3 +330,166 @@ export function assertRequirementsNotTooComplex(
     });
   }
 }
+
+/**
+ * @hidden
+ */
+export async function assertAuthorizationRequestValid(
```

Comment:
We should also validate (if we're not already, which I think is the case) that the authorization still exists by the time it is being accepted (it could have been removed or already accepted before)

Reply:
What I mean is that after accepting or removing an auth, the user still has access to the authorization request object, so they might try to accept it once again

Reply:
That makes sense I will add that in